### ISSUE TITLE
Custom event handlers on programmatic modals

### ISSUE THREAD

**Issue by Tobbe** [rafaelpimpa/buefy#110]
### Overview of the problem

**Buefy** version: 0.4.0
**Vuejs** version: 2.3.4
**OS/Browser**: Win10 Chrome

### Description

Can't add custom event handlers to programmatic modals.

I tried to do it using the `props` property. Might be better to have a dedicated `events` property.

### Steps to reproduce

This doesn't work. Nothing is printed to the console when I emit 'foo' from RemoveUserModal

```javascript
fooMethod() {
    console.log('fooMethod');
},

showRemoveUserModal(email) {
    this.$modal.open({
        component: RemoveUserModal,
        width: 300,
        props: {
            email,
            'v-on:foo': this.fooMethod,
        },
    });
},
```

### Expected behavior

Calling `this.$emit('foo')` inside `RemoveUserModal` should trigger the `fooMethod` function and print 'fooMethod' to the console

### Actual behavior

`fooMethod` is never called

**Comment by rafaelpimpa** [comment]
Gonna take a look at it later, but I'm not sure if it's possible to do it programatically TBH, you could also use the non-programmatic, you have more control that way.

**Comment by christophemarois** [comment]
@Tobbe @rafaelpimpa a workaround for this is to create an event bus (a Vue Instance) and pass it to the modal as props.

So in the `<b-modal>` parent component, you could have:

```vue
<template>
  <b-modal :component="MyComponent" :props="{ eventBus }"></b-modal>
</template>

<script>
  import Vue from 'vue'
  import MyComponent from '@/components/my-component'
  
  export default {
    data () {
      return {
        eventBus: new Vue(),
        MyComponent
      }
    },

    mounted () {
      this.eventBus.$on('my-custom-event', e => alert('called'))
    },

    beforeDestroy () {
      this.eventBus.$off('my-custom-event')
    }
  }
</script>
```

And in `MyComponent.vue`:

```vue
<template>
  <button @click="eventBus.$emit('my-custom-event')">Trigger</button>
</template>

<script>
  export default {
    props: ['eventBus']
  }
</script>
```

This effectively enables parent > modal component, but it isn't as practical as it would be if we could just hook to events by doing `<b-modal @my-custom-event="handler">`. Maybe this event-bus relay could be built in?

**Comment by rafaelpimpa** [comment]
Thanks @christophemarois, gonna take a look at it

**Comment by christophemarois** [comment]
@rafaelpimpa Actually, I did some more tests today, and I think that exploiting Vue's `$parent` object is really the most straightforward way to emit events from a component inside `<b-modal>`.

```vue
<template>
  <b-modal :component="MyComponent" @my-custom-event="myCustomEvent"></b-modal>
</template>

<script>
  import MyComponent from '@/components/MyComponent'
  
  export default {
    data () {
      return {
        MyComponent
      }
    },

    methods: {
      myCustomEvent ($event) {
        console.log($event)
      }
    }
  }
</script>
```

and in `MyComponent.vue`

```vue
<template>
  <!-- Will log "hey" to the console when clicked -->
  <button @click="$parent.$emit('my-custom-event', 'hey')">Trigger</button>
</template>

<script>
  export default {}
</script>
```

**Comment by rafaelpimpa** [comment]
That looks better

**Comment by Tobbe** [comment]
I just implemented the "event bus" solution. Thanks for the idea, works well enough for me, for now 👍

**Comment by rafaelpimpa** [comment]
Closing this since you've got it working. Not sure if I'm gonna change anything.

**Comment by christophemarois** [comment]
Maybe just a heads-up in the modal documentation would do the trick. I agree it is not a problem per se, but rather a Vue-specific mechanic. Thanks for the good work!

**Comment by rafaelpimpa** [comment]
I've updated the modal component from the docs (not published yet) [db66c14](https://github.com/rafaelpimpa/buefy/commit/db66c14c0b4de1285de42de0350f1e557aba4793)

Using like this you have more control over the component:

```html
<b-modal :active.sync="isComponentModalActive" has-modal-card>
    <modal-form v-bind="formProps"></modal-form>
</b-modal>
```
